src/OpenRiaServices.Hosting.AspNetCore/Framework/AspNetCore/OpenRiaServicesOptions.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using Microsoft.AspNetCore.Http;
+using OpenRiaServices.Server;
+
+#nullable enable
+
+namespace OpenRiaServices.Hosting.AspNetCore
+{
+    public sealed class OpenRiaServicesOptions
+    {
+        // TODO: Naming, should it be a single parameter, or should exception details (exception, ServiceContext) be passed as separate parameters?
+        public Action<UnhandledExceptionParameters>? UnhandledException { get; set; }
+        // public OpenRiaServicesOptions WithOnUnhandledException(Action<Exception, DomainServiceErrorInfo>) { }
```

Comment:
I did a bit research on similar naming on other frameworks handle this / 

## ServiceStack

https://docs.servicestack.net/error-handling#overriding-onexceptiontypefilter-in-your-apphost

* There are several interface to put on excetion exceptions (to allow setting result text / error code / http status)
* has a `"OnExceptionTypeFiler(Exception, ResponseStatus)"` to override in app 
 https://docs.servicestack.net/error-handling#overriding-onexceptiontypefilter-in-your-apphost
 (ResponseStatus can have multiple errors

```csharp
responseStatus.Errors.Add(new ResponseError
        {
            ErrorCode = ex.GetType().Name,
            FieldName = argEx.ParamName,
            Message = errorMsg,
        });
```		
		
It also have both ServiceExceptionHandlers 
```csharp
ServiceExceptionHandlers.Add((httpReq, request, exception) => {
        //log your exceptions here
        ...
        return null; //continue with default Error Handling

        //or return your own custom response
        //return DtoUtils.CreateErrorResponse(request, exception);
    });
```	
	
 AND AppHost.UncaughtExceptionHandlers
```csharp
this.UncaughtExceptionHandlersAsync.Add(async (req, res, operationName, ex) =>
{
    await res.WriteAsync($"UncaughtException '{ex.GetType().Name}' at '{req.PathInfo}'");
    res.EndRequest(skipHeaders: true);
});
```

## Microsoft Orleans

Seems to use filters to convert exceptions 

https://learn.microsoft.com/sv-se/dotnet/orleans/grains/interceptors#exception-conversion

## FastEndpoints

https://fast-endpoints.com/docs/exception-handler#unhandled-exception-handler

Has very  baseic functionality (choose how to log and if exception messages should be sent)
https://api-ref.fast-endpoints.com/api/FastEndpoints.ExceptionHandlerExtensions.html

They recommend writing your own exception handler ... :) 
https://github.com/FastEndpoints/FastEndpoints/blob/main/Src/Library/Extensions/ExceptionHandlerExtensions.cs

Reply:
So maybe `UnhandledExceptionHandler` or `UnhandledException`, `UnhandledExceptionCallback` are feasible